commons-db/src/main/java/com/palantir/nexus/db/sql/SQLString.java

```diff
@@ -367,4 +364,28 @@ public String getKey() {
     public static RegisteredSQLString getRegisteredQueryByKey(FinalSQLString key) {
          return new RegisteredSQLString(key.delegate);
     }
+
+    protected static ImmutableMap<String, FinalSQLString> getCachedUnregistered() {
+        synchronized (cacheLock) {
```

Comment:
The GuardedBy comments indicate that we should be able to do a volatile read without locking, is that no longer the case?